include/linux/file_compat.h

```diff
@@ -76,8 +76,17 @@ spl_filp_fallocate(struct file *fp, int mode, loff_t offset, loff_t len)
 #define	spl_filp_fsync(fp, sync)	vfs_fsync(fp, (fp)->f_dentry, sync)
 #endif /* HAVE_2ARGS_VFS_FSYNC */
 
+#ifdef HAVE_INODE_LOCK_SHARED
+#define	spl_inode_lock(ip)		inode_lock_nested(ip, I_MUTEX_PARENT)
+#define	spl_inode_unlock(ip)		inode_unlock(ip)
+#define	spl_inode_lock_shared(ip)	inode_lock_shared(ip)
+#define	spl_inode_unlock_shared(ip)	inode_unlock_shared(ip)
+#else
 #define	spl_inode_lock(ip)		mutex_lock_nested(&(ip)->i_mutex, I_MUTEX_PARENT)
 #define	spl_inode_unlock(ip)		mutex_unlock(&(ip)->i_mutex)
+#define	spl_inode_lock_shared(ip)	mutex_lock(&(ip)->i_mutex)
```

Comment:
For the moment it's fine, but I suspect it will bite us some day.  This raises a related point, it would be prudent to establish `spl_` wrappers now for all of the inode lock variants to avoid future potential spl/zfs compatibility issues.

```
spl_inode_lock
spl_inode_unlock
spl_inode_lock_shared
spl_inode_unlock_shared
spl_inode_trylock
spl_inode_trylock_shared
spl_inode_is_locked
spl_inode_lock_nested
```